src/Nncase.Evaluator/Math/MatMul.cs

```diff
@@ -78,12 +78,16 @@ private IRType Visit(TensorType lhs, TensorType rhs)
             return new TensorType(lhs.DType, new[] { lhs.Shape[0], rhs.Shape[1] });
         }
 
-        var bigShape = lhs.Shape.Rank > rhs.Shape.Rank
-            ? lhs.Shape
-            : rhs.Shape;
+        var lhsShape = lhs.Shape.Rank >= rhs.Shape.Rank ? lhs.Shape.ToArray() : Enumerable.Repeat((Dimension)1, rhs.Shape.Rank - lhs.Shape.Rank).Concat(lhs.Shape).ToArray();
+        var rhsShape = lhs.Shape.Rank <= rhs.Shape.Rank ? rhs.Shape.ToArray() : Enumerable.Repeat((Dimension)1, lhs.Shape.Rank - rhs.Shape.Rank).Concat(rhs.Shape).ToArray();
+
+        var bigShape = Enumerable.Zip(lhsShape, rhsShape).ToArray()[..^2].Select(t =>
```

Comment:
`Enumerable.Zip(lhsShape, rhsShape).Take(2)` is better

Reply:
done